### PR TITLE
check null terminated messages

### PR DESCRIPTION
**Pull request by nickdesaulniers**
https://github.com/nanomsg/nanomsg/pull/363#issuecomment-76783754

review? @reqshark

### CODE REVIEW

File: test/send.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
 // https://github.com/chuckremes/nn-core/blob/master/spec/nn_send_spec.rb
 
-var assert = require('assert');
-var should = require('should');
 var nano = require('../');
 var nn = nano._bindings;
```
Comment (reqshark) [inline test/send.js]: we can remove `var nn = nano._bindings;` here

Comment (nickdesaulniers) [inline test/send.js]: let's leave it for this PR and I'll target unnecessary requires in #74

**Comment by nickdesaulniers** [comment]
huh, bunch of failed tests, I wonder if the utf8 is getting GC'd

**Comment by reqshark** [comment]
osx is passing.

fails on my ubuntu server same as the tests.. hmm

File: src/node_nanomsg.cc [hunk 2]
```diff
@@ -199,30 +199,24 @@ NAN_METHOD(Shutdown) {
...
-
-    const char *data = node::Buffer::Data(object);
-    input = new std::string(data, node::Buffer::Length(object));
-
+  const int flags = args[2].integer;
+  const char* data = NULL;
+  size_t length = 0;
+
+  // check if Local<v8::Value> is a v8::String
+  if (args[1]->IsString()) {
```
Comment (reqshark) [inline src/node_nanomsg.cc]: what's a V8::String?

**Comment by nickdesaulniers** [comment]
I'm quite certain that because I'm using args[1] directly, it's getting GC'd before libuv sends the message, so whatever is sent is completely random memory!  Will need to copy the string. :(

**Comment by nickdesaulniers** [comment]
Also, this only seems to failing for strings, not buffers.

**Comment by reqshark** [comment]
just check for buffer instance otherwise assume string and don't return undefined. that might fix

**Comment by nickdesaulniers** [comment]
pushing another test, not holding my breath though.  I know we have no issues for buffers, but we definitely do for strings, since I believe we're adding the null byte (so this patch still doesn't address the whole point of the PR)

**Comment by nickdesaulniers** [comment]
ah! NO_NULL_TERMINATION

**Comment by nickdesaulniers** [comment]
yep, don't merge yet, I've got a bunch of ideas to clean this up

**Comment by nickdesaulniers** [comment]
so, this whole thing might have been a waste of time.  I just added a test that sends a string, and sure enough it looks like we have nothing to worry about

**Comment by nickdesaulniers** [comment]
that took some serious patch surgery to save just the tests.  These tests help show we don't send null terminated msgs from the application layer.  patch ready for review

**Comment by reqshark** [comment]
LGTM.

btw, I think we may end up encountering some of this issue again in the near future when we introduce zero-copy send operations using the [nn_allocmsg stuff](http://nanomsg.org/v0.5/nn_allocmsg.3.html)

**Comment by reqshark** [comment]
sizzurp 727f552fd0ec095c3beb02c235721e47a7e931f7

**Pull request merged**